performance.lic

```diff
@@ -0,0 +1,156 @@
+custom_require.call(%w(common common-travel))
+  
+class Performance
+  
+  include DRC
+  include DRCT
+  
+  def initialize
+    no_pause_all
+    startTime = Time.new.to_i
+
+    @settings = get_settings
+    @song_list = get_data('perform').perform_options
+    UserVars.song = @song_list.first.first unless UserVars.song
+    fput('release ecry') if DRStats.bard?
+
+    instrument = @settings.instrument
+
+    hand_armor = @settings.hand_armor
+    if instrument != nil
+      bput("get my #{instrument}","You get")
+      bput("remove my #{hand_armor}","You pull off","Remove what")
+      bput("stow my #{hand_armor}","You put","Stow what")
+    end
+    
+    if play_song
+      loop do 
+        runTime = Time.new.to_i - startTime
+        break if runTime > 600
+        break if DRSkill.getxp('Performance') > 30
+        line = get
+        pause 0.05 unless line
+        case line
+        when /You begin (.*)/, /You continue (.*)/, /You find yourself relaxing (.*)/, /You concentrate on your sense of rhythm (.*)/, /You continue (.*)/
+          if DRSkill.getxp('Performance') > 30
+            fput('stop play')
+            exit
+          end
+        when /You finish (.*)/
+          if DRSkill.getxp('Performance') <= 30
+            play_song
+          else
+            exit
+          end
+        when /Perhaps you should find somewhere drier/
+          exit
+        end
+      end
+    end
+     
+    if instrument != nil
+      bput("stow my #{instrument}","You put")
+    end
+     
+  end
+  
+  def play_song
+    
+    @did_play = true
+    result = bput("play #{UserVars.song}", 'dirtiness may affect your performance', 'slightest hint of difficulty', 'fumble slightly', 'Your zills is submerged in the water','You begin a', 'You struggle to begin', 'You\'re already playing a song', 'You effortlessly begin', 'You begin some', 'You cannot play', 'Play on what instrument', 'now isn\'t the best time to be playing', 'Perhaps you should find somewhere drier before trying to play')
+     echo result
```

Comment:
❓ Should this be guarded by a conditional like `if UserVars.something_debug`?

Reply:
Looks like some of my debugging messages were left.  I'll remove them